### ISSUE TITLE
Tuple types with spreads

### ISSUE THREAD

**Issue by basarat** [basarat/typescript-book#353]
Reader comment https://www.gitbook.com/book/basarat/typescript/discussions/72 :rose: Quote: 

> If you want a tuple type like `[string, boolean, ...number]`, you can use `interface YourName extends Array<number> { 0: string, 1: number }`

**Comment by basarat** [comment]
That said, I'm nervous of edge cases. TypeScript has *native* understanding of tuples that might from its understanding of Array at a future point.

**Comment by j-f1** [comment]
I agree. It would be much better to have `[string, boolean, ...number]`-style types built into TS!
